var/spack/repos/builtin/packages/intel-oneapi-mpi/package.py

```diff
@@ -41,6 +43,20 @@ class IntelOneapiMpi(IntelOneApiLibraryPackage):
 
     provides('mpi@:3.1')
 
+    @classmethod
+    def determine_version(cls, exe):
+        """Return the version of the provided executable or ``None`` if
+        the version cannot be determined.
+
+        Arguments:
+            exe (str): absolute path to the executable being examined
+        """
+        versions = [str(v) for v in cls.versions]
+        output = Executable(exe)('--version', output=str, error=str)
```

Comment:
```suggestion
        output = Executable(exe)('-v', output=str, error=str)
```

``--version`` is passed through to the compiler, and gives the compiler version, ``-v`` will show the you the mpi version